src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/EmptyLineSeparatorCheck.java

```diff
@@ -610,7 +610,20 @@ private boolean isPrePreviousLineEmpty(DetailAST token) {
         final int number = 3;
         if (lineNo >= number) {
             final String prePreviousLine = getLine(lineNo - number);
+
             result = CommonUtil.isBlank(prePreviousLine);
+
+            if (token.findFirstToken(TokenTypes.TYPE) != null) {
+                final DetailAST beginningCommentBlock = token.findFirstToken(TokenTypes.TYPE)
+                    .findFirstToken(TokenTypes.BLOCK_COMMENT_BEGIN);
```

Comment:
methods
`private boolean hasEmptyLineBefore(DetailAST token)`
`private boolean isPrePreviousLineEmpty(DetailAST token)`

are almost identiacal, with diference only in 1 digit, please remove copy-paste by usage more arguments